go-controller/pkg/ovn/base_network_controller_namespace.go

```diff
@@ -77,8 +77,9 @@ func (bnc *BaseNetworkController) WatchNamespaces() error {
 		return nil
 	}
 
-	if bnc.IsSecondary() && !util.IsMultiNetworkPoliciesSupportEnabled() {
-		// For secondary networks, we don't have to watch namespace events if
+	if (!bnc.IsPrimaryNetwork() && util.IsNetworkSegmentationSupportEnabled()) &&
```

Comment:
this match got complicated lol specially with the first match on L74 cause we seem to need to do the opposite of that here...

Reply:
I've modified the code to make it more readable.

Reply:
lgtm